packages/js/src/ai-generator/components/modal-content.js

```diff
@@ -64,10 +68,35 @@ export const ModalContent = () => {
 			upsellProps.bundleNote = <div className="yst-text-xs yst-text-slate-500 yst-mt-2">
 				{ `*${upsellPremiumWooLabel}` }
 			</div>;
-			upsellProps.upsellLink = upsellLinkWooPremiumBundle;
+			upsellProps.upsellLink = upsellLinks.bundle;
 			upsellProps.ctbId = "c7e7baa1-2020-420c-a427-89701700b607";
 		}
 	}
+	return upsellProps;
+};
+
+/**
+ * @returns {JSX.Element} The element.
+ */
+export const ModalContent = () => {
+	const upsellLinks = {
+		premium: useSelect( select => select( STORE ).selectLink( "https://yoa.st/ai-generator-upsell" ), [] ),
+		bundle: useSelect( select => select( STORE ).selectLink( "https://yoa.st/ai-generator-upsell-woo-seo-premium-bundle" ), [] ),
+		woo: useSelect( select => select( STORE ).selectLink( "https://yoa.st/ai-generator-upsell-woo-seo" ), [] ),
+	};
+
+	const upsellProps = getUpsellProps( upsellLinks );
+
+	// Use specific copy for product posts.
+	const isWooCommerceActive = useSelect( select => select( STORE ).getIsWooCommerceActive(), [] );
+	const isProductPost = useSelect( select => select( STORE ).getIsProduct(), [] );
+
+	if ( isWooCommerceActive && isProductPost ) {
+		upsellProps.title = __( "Generate product titles & descriptions with AI!", "wordpress-seo" );
+		upsellProps.isProductCopy = true;
+	}
```

Comment:
I moved this out of the `getUpsellProps` function as this is specific for AI Generate.